src/Storages/MergeTree/MergeTreeIndexText.cpp

```diff
@@ -618,43 +641,203 @@ void serializeTokensRaw(
  * 1. https://doi.org/10.1109/Innovate-Data.2017.9
  * 2. https://doi.org/10.1145/3448016.345279
  */
+template <typename TokenGetter>
 void serializeTokensFrontCoding(
-    SerializationStats & stats,
-    WriteBuffer & write_buffer,
-    const SortedTokensAndPostings & tokens_and_postings,
+    const TokenGetter & token_getter,
+    WriteBuffer & ostr,
     size_t block_begin,
     size_t block_end)
 {
-    const auto & first_token = tokens_and_postings[block_begin].first;
-    writeVarUInt(first_token.size(), write_buffer);
-    write_buffer.write(first_token.data(), first_token.size());
+    const auto & first_token = token_getter(block_begin);
+    writeVarUInt(first_token.size(), ostr);
+    ostr.write(first_token.data(), first_token.size());
 
     std::string_view previous_token = first_token;
     for (size_t i = block_begin + 1; i < block_end; ++i)
     {
-        auto current_token = tokens_and_postings[i].first;
+        auto current_token = token_getter(i);
         auto lcp = computeCommonPrefixLength(previous_token, current_token);
-        writeVarUInt(lcp, write_buffer);
-        writeVarUInt(current_token.size() - lcp, write_buffer);
-        write_buffer.write(current_token.data() + lcp, current_token.size() - lcp);
+        writeVarUInt(lcp, ostr);
+        writeVarUInt(current_token.size() - lcp, ostr);
+        ostr.write(current_token.data() + lcp, current_token.size() - lcp);
         previous_token = current_token;
+    }
+}
+
+/// Split postings into smaller blocks without copying the data.
+/// We use the fact that the Roaring Bitmap is split into small
+/// containers that are are stored in contiguous memory and sorted
+/// by the key. Therefore, to create a view to the smaller bitmap,
+/// we need only to adjust the pointers to the containers.
+std::vector<roaring::api::roaring_bitmap_t> splitPostings(const roaring::api::roaring_bitmap_t & postings, size_t block_size)
+{
+    std::vector<roaring::api::roaring_bitmap_t> result;
```

Comment:
[minor] since `block_size` can be configurable, the result array size may vary and can be big. We can allocate enough space as:
```cpp
result.reserve((postings.cardinality() + block_size - 1) / block_size);
```